std/algorithm/sorting.d

```diff
@@ -1048,147 +1048,275 @@ unittest
+            {
+                if (!source[i].empty)
+                {
+                    if (bestIndex == size_t.max || // either this is the first or
+                        comp(bestElement, source[i].back))
+                    {
+                        bestIndex = i;
+                        bestElement = source[i].back;
+                    }
+                }
+            }
+            return bestIndex;
         }
-        assert(false);
-    }
 
-    ///
-    @property auto ref ElementType front()
-    {
-        assert(!empty);
-        // Assume _crt is correct
-        foreach (i, U; Rs)
+        void popBack()
         {
-            if (i < _crt) continue;
-            assert(!_r[i].empty);
-            return _r[i].front;
+            final switch (_lastBackIndex)
+            {
+                foreach (i, _; Rs)
+                {
+                case i:
+                    source[i].popBack();
+                    break;
+                }
+            }
+            _lastBackIndex = backIndex;
         }
-        assert(false);
     }
 
-    static if (allSatisfy!(isForwardRange, Rs))
+    static if (allSatisfy!(isForwardRange, staticMap!(Unqual, Rs)))
     {
-        ///
         @property auto save()
         {
-            auto ret = this;
-            foreach (ti, elem; _r)
+            auto result = this;
+            foreach (i, _; Rs)
             {
-                ret._r[ti] = elem.save;
+                result.source[i] = result.source[i].save;
             }
-            return ret;
+            return result;
         }
     }
 
     static if (allSatisfy!(hasLength, Rs))
     {
-        ///
         @property size_t length()
         {
             size_t result;
-            foreach (i, U; Rs)
+            foreach (i, _; Rs)
             {
-                result += _r[i].length;
+                result += source[i].length;
             }
             return result;
         }
 
-        ///
         alias opDollar = length;
     }
+
+    public Rs source;
+    private size_t _lastFrontIndex = size_t.max;
+    static if (isBidirectional)
+    {
+        private size_t _lastBackIndex = size_t.max;
+    }
 }
 
-/// Ditto
-Merge!(less, Rs) merge(alias less = "a < b", Rs...)
-(Rs rs)
+/**
+   Merge multiple sorted ranges `rs` with less-than predicate function `pred`
+   into one single sorted output range containing the sorted union of the
+   elements of inputs. Duplicates are not eliminated, meaning that the total
+   number of elements in the output is the sum of all elements in the ranges
+   passed to it; the `length` member is offered if all inputs also have
+   `length`. The element types of all the inputs must have a common type
+   `CommonType`.
+
+Params:
+    less = Predicate the given ranges are sorted by.
+    rs = The ranges to compute the union for.
+
+Returns:
+    A range containing the union of the given ranges.
+
+See_Also:
+   $(REF algorithm, setops, SetUnion)
```

Comment:
REF has a different order of arguments than XREF. Symbol name goes first, followed by the module including std. Here: `REF(SetUnion, std,algorithm,setops)`. However, there doesn't seem to be a std.algorithm.setops.SetUnion.